src/test/java/unit/org/apache/tomcat/vault/VaultToolTest.java

```diff
@@ -0,0 +1,142 @@
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.io.ByteArrayOutputStream;
+import java.io.PrintStream;
+import java.lang.reflect.Field;
+import java.security.Permission;
+import static org.junit.Assert.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class VaultToolTest {
+
+    SecurityManager sm = System.getSecurityManager();
+
+    //New exception that we need to catch to prevent methods to exit
+    protected static class ExitException extends SecurityException
+    {
+        public final int status;
+        public ExitException(int status)
+        {
+            super(Integer.toString(status));
+            this.status = status;
+        }
+    }
+
+    //New security manager needed to prevent System.exit
+    private static class NoExitSecurityManager extends SecurityManager
+    {
+        @Override
+        public void checkPermission(Permission perm)
+        {
+            // allow anything.
+        }
+        @Override
+        public void checkPermission(Permission perm, Object context)
+        {
+            // allow anything.
+        }
+        @Override
+        public void checkExit(int status)
+        {
+            super.checkExit(status);
+            throw new ExitException(status);
+        }
+    }
+
+    @Before
+    public void setUp()
+    {
+        System.setSecurityManager(new NoExitSecurityManager());
+    }
+
+
+    @After
+    public void tearDown()
+    {
+        System.setSecurityManager(sm);
+    }
+
+
+    // Helper method to access private fields using reflection
+    private Object getPrivateField(Object instance) throws NoSuchFieldException, IllegalAccessException {
+        Field field = instance.getClass().getDeclaredField("cmdLine");
+        field.setAccessible(true);
+        return field.get(instance);
+    }
+
+
+    // Test: Parse valid arguments successfully
+    @Test
+    public void testValidArguments() throws NoSuchFieldException, IllegalAccessException {
+        String[] args = {"-k", "keystore.jks", "-p", "password", "-x", "secure-value"};
+        VaultTool vaultTool = new VaultTool(args);
+
+        CommandLine cmdLine = (CommandLine) getPrivateField(vaultTool);
+
+        assertTrue(cmdLine.hasOption("k"));
+        assertEquals("keystore.jks", cmdLine.getOptionValue("k"));
+
+        assertTrue(cmdLine.hasOption("p"));
+        assertEquals("password", cmdLine.getOptionValue("p"));
+
+        assertTrue(cmdLine.hasOption("x"));
+        assertEquals("secure-value", cmdLine.getOptionValue("x"));
+    }
+
+    // Test: Missing required argument throws the correct error
+    @Test
+    public void testMissingArgument(){
+
+        String[] args = {"-k", "keystore.jks", "-p"};
+        //int statusCode = catchSystemExit(() -> new VaultTool(args));
+        try{
+            new VaultTool(args);
+        }catch(ExitException e){
```

Comment:
I would prefer spaces around keywords and expressions.

```java
try {
    new VaultTool(args);
} catch (ExitException e) {
````

Just to be consistent with the rest. (It's on multiple places)